### ISSUE TITLE
Standard Keyboard Functionality: Shift + End/Home doesn't work

### ISSUE THREAD

**Issue by derekperkins** [Microsoft/vscode#10807]
- VSCode Version: 1.4
- OS Version: MacOS 10.11.6

Steps to Reproduce:
1. Shift + End doesn't select from the current cursor location to the end of the line.
2. Shift + Home doesn't select from the current cursor location to the beginning of the line.

**Comment by AnthonyMichaelc** [comment]
Have you gone into your preferences  and then click on keyboard shortcuts? when you click that it will bring up the shortcuts and a page for keybindings that you can set those two functions and override the default. Here is a link to the list of keyboard functions as well [https://code.visualstudio.com/docs/customization/keybindings](url)

**Comment by derekperkins** [comment]
Those commands are in the default bindings. The alternate `shift+cmd+right` works, but `shift+end` does not work, at least for me. It also doesn't work if I paste them into my user settings, which should override any extensions if those were overriding the built ins. 

```
{ "key": "shift+cmd+right",       "command": "cursorEndSelect",
                                     "when": "editorTextFocus" },
{ "key": "shift+end",             "command": "cursorEndSelect",
                                     "when": "editorTextFocus" },
...
{ "key": "shift+cmd+left",        "command": "cursorHomeSelect",
                                     "when": "editorTextFocus" },
{ "key": "shift+home",            "command": "cursorHomeSelect",
                                     "when": "editorTextFocus" },
```

**Comment by derekperkins** [comment]
And to clarify, `shift+end` and `shift+home` work in both Sublime and Atom on my machine, so I don't think there's an issue with my keyboard not registering correctly.

**Comment by sandy081** [comment]
@derekperkins `shift+end` and `shift+home` keyboard shortcuts are working as expected for me. 

On Mac I used following keys for

end : `fn + right arrow`
home: `fn + left arrow`

Can you please try with these and let me know if it is still not working.

Thanks

**Comment by derekperkins** [comment]
That combo worked for me, but an external keyboard with physical End/Home buttons does not work when combined with Shift. Inside VSCode, I can use Home and End and they work as expected, but Shift + End or Shift + Home does nothing. Those same commands do work in Sublime, Atom, Microsoft Word, etc, so correct keybindings are being sent.

**Comment by alexandrudima** [comment]
I have validated that we now (i.e. after PR #22894 aka #17521 -- in today's Insiders build) appear to behave correctly for these keys. I have attached a standard QWERTY keyboard to a mac mini and configured OSX to use the US layout. Pressing the Home/End keys appear to be interpreted correctly:


![image](https://cloud.githubusercontent.com/assets/5047891/24298564/a4f12fcc-10a6-11e7-8aa9-7540154f7046.png)
